dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/NonPersistedDataApi.kt

```diff
@@ -0,0 +1,36 @@
+package org.dataland.datalandbackend.api
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import io.swagger.v3.oas.annotations.responses.ApiResponses
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestMapping
+
+/**
+ * Defines the restful dataland-backend API regarding internal data exchange
+ */
+@RequestMapping("/internal/nonpersisted")
+interface NonPersistedDataApi {
+
+    /**
+     * This method retrieves data entries from the temporary storage
+     * @param dataId filters the requested data to a specific entry.
+     */
+    @Operation(
+        summary = "Retrieve specific data from the hashmap of the backend.",
+        description = "Data identified by the provided data ID is retrieved.",
+    )
+    @ApiResponses(
+        value = [
+            ApiResponse(responseCode = "200", description = "Successfully retrieved data set."),
+        ],
+    )
+    @GetMapping(
+        value = ["/{dataId}"],
+        produces = ["application/json"],
+    )
+    fun getCompanyAssociatedDataForInternalStorage(@PathVariable("dataId") dataId: String):
```

Comment:
https://github.com/d-fine/Dataland/pull/180/files#r1114025538 kind of applies here as well:
The part `ForInternalStorage` should - in my view - be avoided. The whole Queue-Integration-Pattern gets it's value from inversion of execution flows: It's not the backend that tells somebody else to do something. It's the backend that tells that something happened - and other involved parties see the event, and know how to react. 

The services are independent Agents - the need no orders, they work on their own. They don't need control.

Reply:
But I agree here. And the change is almost pushed.

Reply:
ok